### PR TITLE
Revert "Fix magic projectiles mirror"

### PR DESCRIPTION
**Pull request by Manly-Dwarf**
Reverts HippieStation/HippieStation#12127
Broke tasers.
This PR is only here so the maintainers will actually see that they broke tasers.

### CODE REVIEW

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -50,13 +50,11 @@
...
 	var/armor = run_armor_check(def_zone, P.flag, "","",P.armour_penetration)
-	var/on_hit_state = P.on_hit(src, armor)
-	if(!P.nodamage && on_hit_state != BULLET_ACT_BLOCK)
+	if(!P.nodamage)
 		apply_damage(P.damage, P.damage_type, def_zone, armor)
-		apply_effects(P.stun, P.knockdown, P.unconscious, P.irradiate, P.slur, P.stutter, P.eyeblur, P.drowsy, armor, P.stamina, P.jitter, P.paralyze, P.immobilize)
 		if(P.dismemberment)
 			check_projectile_dismemberment(P, def_zone)
-	return on_hit_state ? BULLET_ACT_HIT : BULLET_ACT_BLOCK
+	return P.on_hit(src, armor)? BULLET_ACT_HIT : BULLET_ACT_BLOCK
```
Comment (Manly-Dwarf) [inline code/modules/mob/living/living_defense.dm:57]: Either this

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -208,7 +208,7 @@
 	else
 		L.log_message("has been shot by [firer] with [src]", LOG_ATTACK, color="orange")
 
-	return BULLET_ACT_HIT
+	return L.apply_effects(stun, knockdown, unconscious, irradiate, slur, stutter, eyeblur, drowsy, blocked, stamina, jitter, paralyze, immobilize)
```
Comment (Manly-Dwarf) [inline code/modules/projectiles/projectile.dm:211]: or this broke tasers

**Review by Manly-Dwarf — COMMENTED** [review]
heeelp

**Comment by SireTurret** [comment]
then yoyo and steamport get surprised when we tell them to stop merging pr's from tg station

or when we start seething when they try to bring back the mirror bot to port "fixes" (that PR was a fix by the way)

**Comment by Manly-Dwarf** [comment]
oy turret how about instead of shittin on the maintainers you try to help em out
not exactly encouraging to fix code when you only shit on them when they do wrong, no thanks when they do right, eh?

**Comment by SireTurret** [comment]
i ain't really shitting on them, just seethin a bit because of this after they tried to bring back the mirror bot "which would've only ported fixes"

tasers getting broken is something pretty major too

**Comment by SireTurret** [comment]
i ain't really shitting on them, just seethin a bit because of this after they tried to bring back the mirror bot "which would've only ported fixes"

tasers getting broken is something pretty major too

File: code/modules/mob/living/living_defense.dm [hunk 3]
```diff
@@ -50,13 +50,11 @@
 
 /mob/living/bullet_act(obj/item/projectile/P, def_zone)
 	var/armor = run_armor_check(def_zone, P.flag, "","",P.armour_penetration)
-	var/on_hit_state = P.on_hit(src, armor)
-	if(!P.nodamage && on_hit_state != BULLET_ACT_BLOCK)
+	if(!P.nodamage)
```
Comment (ghost) [inline code/modules/mob/living/living_defense.dm:53]: Tasers have nodamage set to TRUE so this always fails for them.
Tasers only apply a status effect and don't modify any health values of the mob

Disabler beams do not override nodamage to being true so they apply their stamina damage and any status effects.

File: code/modules/mob/living/living_defense.dm (hunk 1 above)
Comment (Manly-Dwarf) [reply code/modules/mob/living/living_defense.dm:57]: github advertises multiline reviews and it doesn't even work

**Comment by diegoxt45** [comment]
dwarf gave me brain damage for this so i'm gonna +1 it

**Comment by yoyobatty** [comment]
Nah this isn't a case of tg ruins things, this is a case of oops, forgot that something else forgot to get merged with this. An oversight, don't need to make it more complicated and start pointing fingers like a bunch of 10 year olds.

**Pull request closed without merging**